types/truffle-hdwallet-provider/index.d.ts

```diff
@@ -0,0 +1,19 @@
+// Type definitions for truffle-hdwallet-provider 0.0
+// Project: https://github.com/trufflesuite/truffle-hdwallet-provider#readme
+// Definitions by: Leonid Logvinov <https://github.com/LogvinovLeon>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.4
+
+import {
+    JSONRPCRequestPayload,
+    JSONRPCResponsePayload,
+    Provider
+} from "ethereum-protocol";
+declare class HDWalletProvider implements Provider {
+    constructor(mnemonic: string, rpcUrl: string);
+    sendAsync(
+        payload: JSONRPCRequestPayload,
+        callback: (err: Error, result: JSONRPCResponsePayload) => void
+    ): void;
+}
```

Comment:
There is no documentation for sendAsync and in https://github.com/trufflesuite/truffle-hdwallet-provider/blob/master/index.js there are more methods than that need to be added if this is added. Cannot verify type from that though so would assume you are typing it correctly.

Reply:
So are send, getAddress, getAddresses suppose to be private then? Probably add those as private then? I am just looking at the code and seeing if your type reflects it or not.